rust/azure_iot_operations_mqtt/src/azure_mqtt_adapter.rs

```diff
@@ -0,0 +1,575 @@
+            let sat_auth = fs::read(sat_file).map_err(|e| ConnectionSettingsAdapterError {
+                msg: "cannot read sat auth file".to_string(),
+                field: ConnectionSettingsField::SatAuthFile(sat_file.clone()),
+                source: Some(Box::new(e)),
+            })?;
+            Ok(Some(AuthenticationInfo {
+                method: "K8S-SAT".to_string(),
+                data: Some(sat_auth.into()),
+            }))
+        } else {
+            Ok(None)
+        }
+    }
+}
+
+impl MqttConnectionSettings {
+    /// Convert to Azure MQTT connect parameters
+    ///
+    /// # Parameters
+    /// - `user_properties`: User properties to include in the connect properties
+    /// - `outgoing_max`: Outgoing message queue size
+    ///
+    /// # Errors
+    /// Returns [`ConnectionSettingsAdapterError`] if any conversion fails
+    pub fn to_azure_mqtt_connect_parameters(
+        self,
+        user_properties: Vec<(String, String)>,
+        outgoing_max: usize,
+    ) -> Result<
+        (
+            azure_mqtt::client::ClientOptions,
+            AzureMqttConnectParameters,
+        ),
+        ConnectionSettingsAdapterError,
+    > {
+        let client_options = ClientOptions {
+            client_id: Some(self.client_id),
+            queue_size: outgoing_max,
+        };
+
+        let connect_options =
+            create_connect_options(self.username, self.password, self.password_file)?;
+
+        let connect_properties = create_connect_properties(
+            self.session_expiry,
+            self.receive_packet_size_max,
+            self.receive_max,
+            user_properties,
+        )?;
+
+        let connection_transport_config = create_connection_transport_config(
+            self.ca_file,
+            self.cert_file,
+            self.key_file,
+            self.key_password_file,
+            self.use_tls,
+            self.hostname,
+            self.tcp_port,
+        )?;
+
+        Ok((
+            client_options,
+            AzureMqttConnectParameters {
+                initial_clean_start: self.clean_start,
+                keep_alive: self.keep_alive,
+                connection_transport_config,
+                connect_options,
+                connect_properties,
+                sat_file: self.sat_file,
+            },
+        ))
+    }
+}
+
+fn read_root_ca_certs(ca_file: String) -> Result<Vec<X509>, anyhow::Error> {
+    let mut ca_certs = Vec::new();
+    let ca_pem = fs::read(ca_file)?;
+
+    let certs = &mut X509::stack_from_pem(&ca_pem)?;
+    ca_certs.append(certs);
+
+    if ca_certs.is_empty() {
+        Err(TlsError::new("No CA certs available in CA File"))?;
+    }
+
+    ca_certs.sort();
+    ca_certs.dedup();
+
+    Ok(ca_certs)
+}
+
+fn tls_config(
+    ca_file: Option<String>,
+    cert_file: Option<String>,
+    key_file: Option<String>,
+    key_password_file: Option<String>,
+) -> Result<(Option<ClientCert>, Vec<X509>), anyhow::Error> {
```

Comment:
can you add a comment documenting what these return types represent? (client_cert, ca_trust_bundle)
Just for readability in the future